### ISSUE TITLE
Rejects definition of super-table of table array

### ISSUE THREAD

**Issue by jorisvr** [marzer/tomlplusplus#187]
## The non-conforming TOML snippet
```toml
[[a.b]]
x = 1

[a]
y = 2
```

This testcase is part of `toml-test`: https://github.com/BurntSushi/toml-test/blob/master/tests/valid/table/array-implicit-and-explicit-after.toml

## What you expected
This is a valid TOML document which should be parsed as
`{"a": {"b": [{"x": 1}], "y": 2}}`

The TOML 1.0.0 spec gives the following example as valid:
```toml
[x.y.z.w]
[x] # defining a super-table afterward is ok
```
There is no reason why an implicit super-table should be treated differently when created by a table-array definition vs a table definition. Nothing in the TOML spec suggests that super-tables of table arrays should be considered as _defined_ instead of merely implicitly _created_.

## What you got
```
Error while parsing table header: cannot redefine existing table 'a'
     	(error occurred at line 4, column 1 of 'stdin')
```

## Environment
**toml++ version and/or commit hash:**  
version v3.2.0, commit 698285d9b2f3f6756fcdab8b93f60352325764e1 on master branch

**Any other useful information:**

**Comment by marzer** [comment]
Ah, thanks for the report. I'll look into this later today :)

**Comment by marzer** [comment]
Sorry, ran out of time. Will have a look during the week.

**Comment by jorisvr** [comment]
Thanks for looking into it. Please take the time you need.

To be clear, this issue is not blocking me in any way. I just happened to notice this and created an issue because getting this fixed may be good for your project in the long run. But it is of course up to you to decide whether this has any priority.

**Comment by marzer** [comment]
I've fixed this in 3.3.0, thanks again for the report!
